java/androidpayload/library/src/com/metasploit/meterpreter/IntervalCollectionManager.java

```diff
@@ -0,0 +1,112 @@
+package com.metasploit.meterpreter;
+
+import android.content.Context;
+
+import java.util.Enumeration;
+import java.util.Hashtable;
+
+public class IntervalCollectionManager {
+
+    private static final int COLLECT_TYPE_WIFI = 1;
+
+    private final Context context;
+    private final Hashtable<Integer, IntervalCollector> collectors;
+
+    public IntervalCollectionManager(Context context) {
+        this.context = context;
+        this.collectors = new Hashtable<Integer, IntervalCollector>();
+    }
+
+    public boolean createCollector(int type, long timeout) {
+        IntervalCollector collector = this.getCollector(type);
+
```

Comment:
Rather than having a hardcoded switch statement, would it be better to declare an array of type/constructor pairs and then construct the new one generically if it's in the list? If we're going to maybe support loading new collectors as extensions, that would give them a way to register with createCollector